### PR TITLE
ffmpeg: add tensorflow dnn backend support

### PR DESCRIPTION
**Pull request by aviallon**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] (_ongoing_) Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aviallon** [comment]
@Atemu should `withTensorflow` be enabled when `withFullDeps` is set?

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 1]
```diff
@@ -79,6 +79,7 @@
 , withSsh ? withHeadlessDeps # SFTP protocol
 , withSvg ? withFullDeps # SVG protocol
 , withSvtav1 ? withFullDeps && !stdenv.isAarch64 # AV1 encoder/decoder (focused on speed and correctness)
+, withTensorflow ? withFullDeps # Tensorflow dnn backend support
```
Comment (Atemu) [inline pkgs/development/libraries/ffmpeg/generic.nix:82]:
```suggestion
, withTensorflow ? false # Tensorflow dnn backend support
```

**Review by Atemu — COMMENTED** [review]
Builds fine for me. Not sure how I could test this exactly.

Given that TF is marked as insecure and is _really_ fat, I don't think ffmpeg_full should have it. If it's unmarked in the future, we can think about a ffmpeg_fat variant. Though perhaps we could make the regular ffmpeg fatter (allowing commonly used things like nvenc, vaapi, AV1 etc.) such that running ffmpeg_full is less of a necessity. We'll cross that bridge when we get there though.

**Comment by Atemu** [comment]
(Noticed https://github.com/NixOS/nixpkgs/pull/225464 while reviewing this. Whoops.)

**Review by Atemu — APPROVED** [review]
Please target staging, otherwise LGTM.

**Comment by aviallon** [comment]
@Atemu targeted staging.

**Pull request merged**